examples/hf_serialization.py

```diff
@@ -39,15 +61,43 @@
 logger.addHandler(fh)
 
 
+def check_file_exists(
+    file: str,
+):
+    """
+    Check if file exists and is not empty. If the file is found locally,
+    it is checked for emptiness with `os.path.exists`. If the file is found
+    on S3, it is checked for emptiness by reading the first byte of the file.
+
+    Args:
+        file: The path to check for existence and emptiness. This can
+           either be a local path or an S3 URI.
+    """
+    if os.path.exists(file):
+        return True
+    else:
+        try:
+            with stream_io.open_stream(
+                file,
+                "rb",
+                *s3_read_credentials,
+            ) as f:
+                f.read(1)
+                return True
+        except OSError as e:
+            return False
+
+
 def serialize_model(
     model: torch.nn.Module,
     config: Optional[Union[ConfigMixin, AutoConfig, dict]],
     model_directory: str,
-    model_prefix: str = "model",
+    model_prefix: str = None,
```

Comment:
If the default is actually `"model"`, then it should be set as a default in the signature for clarity (rather than checking against `None` only to set it to `"model"` later)
```suggestion
    model_prefix: str = "model",
```

Reply:
That's what I'd typically do, but I got a bit confused from the original handling in the source code for `load_model` that I thought reflected a general design preference:

```
def load_model(
    path_uri: str,
    model_class: Union[
        Type[PreTrainedModel], Type[ModelMixin], Type[ConfigMixin]
    ],
    config_class: Optional[
        Union[Type[PretrainedConfig], Type[ConfigMixin], Type[AutoConfig]]
    ] = None,
    model_prefix: Optional[str] = "model",
    device: torch.device = utils.get_device(),
    dtype: Optional[str] = None,
) -> torch.nn.Module:
    """
    Given a path prefix, load the model with a custom extension

    Args:
        path_uri: path to the model. Can be a local path or a URI
        model_class: The model class to load the tensors into.
        config_class: The config class to load the model config into. This must be
            set if you are loading a model from HuggingFace Transformers.
        model_prefix: The prefix to use to distinguish between multiple serialized
            models. The default is "model".
        device: The device onto which to load the model.
        dtype: The dtype to load the tensors into. If None, the dtype is inferred from
            the model.
    """

    if model_prefix is None:
        model_prefix = "model"
```

As I wasn't sure how this was meant to pass `None`, outside of explicitly putting `None` as its positional argument. I think I was just confused and I agree with your suggestion so happy to commit this.

Reply:
If the type on it is `Optional[str]`, that means that it's valid to put `None` as a positional argument. In this case, it makes more sense for it to be `str`, since the default can be clearly defined in the signature. Using `None` as a default makes sense if it's dynamic or based on other arguments in a way that can't be embedded in the call signature.

One reason why functions would be written like this otherwise is to not have to remember the default value if the calling code may or may not have a value to put there, and could conditionally need a default. In that case, the default would probably be better set further up the call chain, though.